Hashiconf docs changes (#2178)
hashicorp/boundary: website/content/docs/getting-started/deploy-and-login.mdx

---
layout: docs
page_title: Deploy and Login
description: |-
  How to deploy HCP Boundary services and login for the first time.
---

> # Deploy an HCP Boundary Instance

Reviewer: This section should include the steps format.

> Navigate to HashiCorp Cloud Platform and login using your credentials. Click on

Reviewer: ```suggestion
1. Navigate to HashiCorp Cloud Platform and login using your credentials. 
2. Click 
```

> the **Boundary (Beta)** tab and click **Deploy Boundary** on the right-hand

Reviewer: ```suggestion
the **Boundary (Beta)** tab, and then and click **Deploy Boundary** on the right-hand
```

pane. 

> Fill out the following form details:

Reviewer: ```suggestion
3. Fill out the following form details:
```

- Instance Name
- Administrator Username
- Administrator Password

> Store the administrator username and password in a safe location. These

Reviewer: ```suggestion
4. Store the administrator username and password in a safe location. These
```

credentials will be used when authenticating using the Admin Console, CLI, or
Terraform.

> Click **Deploy** after completing the form. It may take a few minutes to

Reviewer: ```suggestion
5. Click **Deploy** after completing the form. It may take a few minutes to
```

initialize the HCP Boundary instance.

![Deploy boundary](/img/hcp-deploy-form.png)

## Login to HCP Boundary

Once the deployment is complete, the HCP Boundary instance can be accessed
directly from the HCP Boundary portal using the credentials created during
setup. 

> From the HCP Boundary portal page, click the **Open Admin UI** button. 

Reviewer: ```suggestion
1. From the HCP Boundary portal page, click the **Open Admin UI** button. 
```

> Enter the admin username and password you created when you deployed the new

Reviewer: ```suggestion
2. Enter the admin username and password you created when you deployed the new
```

instance and click **Authenticate**.

The **Welcome to Boundary** setup wizard will launch automatically. There are
two options for Boundary’s initial setup:

> 1. Quick setup

Author: there seems to be an issue with the markdown in getting an unordered list to indent correctly under an ordered list. Will log this for the future to try and find a solution.

1. Manual

![Setup Wizard](/img/setup-wizard.png)

The **Quick setup** is recommended for new users, and generates a set of initial
resources to assist in connecting to your first target. 

![Setup Wizard](/img/setup-wizard-target.png)

The **Manual** setup drops the user directly into Boundary, without any
provisioned resources.

> Upon completion of the setup wizard you are logged into your HCP Boundary

Reviewer: ```suggestion
Upon completion of the setup wizard, you will be logged into your HCP Boundary
```

instance's Global scope via the web UI. This is the default scope for all new
Boundary clusters.

> Click **Auth Methods**.

Reviewer: ```suggestion
3. Click **Auth Methods**.
```

Click the **copy icon** for the Password **Auth Method ID**. The ID is needed to
authenticate to HCP Boundary using the CLI.

<Tabs>
<Tab heading="CLI">

In the HCP Boundary portal, open the **Boundary Overview** page and click the
**copy icon** in the **Copy this into Boundary Desktop** section. This copies
your environment’s Boundary **Origin URL**.

Open a terminal session and set the `BOUNDARY_ADDR` environment variable to the
copied origin URL.

```shell-session
$ export BOUNDARY_ADDR=<boundary-origin-url>
```

Log in with the administrator credentials you created when you deployed the HCP
Boundary instance.

```shell-session
$ boundary authenticate password \
   -auth-method-id=COPIED_AUTH_ID \
   -login-name=ADMIN_USERNAME \
   -password=ADMIN_PASSWORD
```

You are now logged into your HCP Boundary instance's Global scope via the CLI.
This is the default scope for all new Boundary clusters.

If you are on Unix-like operating system (other than macOS/Darwin), you may get

> an error indicating that the token could not be stored, as the freedesktop.org

Reviewer 2: DBUS-based secret service stuff is pretty problematic and very often doesn't work in terminal sessions. I recommend it only for those that know what they're doing. The default on *nix systems is `pass` and is the recommended approach.

Author: I'd appreciate if you could author an alternative bit of text if you want this updated, since it's copied directly from the existing [run and login docs](https://www.boundaryproject.io/docs/getting-started/run-and-login#login-to-boundary).

Reviewer 2: I'm not sure how you'd want to construct this. Would you want to point people to pass? Do you want to include any setup or configuration information for it? Would you want any other text as a backup, or to mention that alternatives are available?

Secret Service implementation is not always available. On these systems, you can
work around this by installing `dbus-x11` and `gnome-keyring` using your package
manager, then creating and unlocking the default keyring with the following,
substituting in a password of your choice for "foobar" (but ending with `\n`).
You can also avoid putting the password on the command line by running the
`gnome-keyring-daemon` commands directly and entering in the password, followed
by a newline (return) and an EOF (`Ctrl+D`).

> - `eval "$(printf 'foobar\n' | gnome-keyring-daemon --unlock)"`

Reviewer 2: This is all quite complicated and much simpler to just use `pass`.

Author: This is the existing solution in the docs, would be happy to collaborate on an alternative. This was added due to the many comments in discuss regarding the keyring issue

Reviewer 2: Unfortunately it's not really "the keyring issue" as every different keyring can have its own issue as they are different implementations. I'm not sure which discuss threads you're referring to but if they're relatively old they probably predate `pass` being made the default (or, possibly, even supported at all).

We should definitely not have this be the promoted solution. I didn't realize the docs had this in here at all. It also doesn't match the default behavior of the CLI which gives errors/warnings about `pass` if it doesn't work.

Author: what solution would you promote instead? Is documenting a solution for this issue even necessary with the implementation of `pass`?

- `eval "$(printf 'foobar\n' | gnome-keyring-daemon --start)"`

This would have to be run in each shell.

If you're unable to install these packages, or don't want to, you can tell the
Boundary authenticate command to not save the token to the operating system's
key manager by setting `-token-name=none` flag or `BOUNDARY_TOKEN_NAME=none` env
variable when running `boundary authenticate`. You'll be responsible for setting
the token in subsequent commands via `-token` flag or `BOUNDARY_TOKEN` env
variable. An easy way to do this would be to use the `-format=json` flag along
with `jq` to pull the token value out of the response and place it wherever you
wish, then create a command alias for `boundary` that sources that value into
the environment or the `-token` flag.

> ~> Token storage on \*nix systems has been more problematic than we expected.

Reviewer 2: This issue is closed and we shouldn't encourage people to comment on it. In addition this ticket is outdated; it was created before we settled on `pass` as the default.

Author: thanks for the context! Let's sync on the alternative `pass` method.
